src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LightCompiler.cs

```diff
@@ -2790,7 +2790,11 @@ private void CompileTypeEqualExpression(Expression expr)
             var node = (TypeBinaryExpression)expr;
 
             Compile(node.Expression);
-            if (node.TypeOperand.GetTypeInfo().IsGenericType && node.TypeOperand.GetGenericTypeDefinition() == typeof(Nullable<>))
+            if (node.Expression.Type == typeof(void))
+            {
+                _instructions.EmitLoad(node.TypeOperand == node.Expression.Type, typeof(bool));
+            }
+            else if (node.TypeOperand.GetTypeInfo().IsGenericType && node.TypeOperand.GetGenericTypeDefinition() == typeof(Nullable<>))
```

Comment:
Probably. I just added the extra case rather than looked at any clean-up. I came across a few other bits that could be made a bit cleaner, but I neither want to add them to my latests PRs and add to their complexity, nor add a separate PR that will conflict with these if merged. I'm going to hang back on that until these are either merged or rejected.